### PR TITLE
docs: Update Modal examples

### PR DESCRIPTION
**Pull request by emroller16**
## Summary

Fixes: #1620

Currently, our Modal Documentation page on Canvas is lacking documentation and just shows examples from Storybook. Since all the examples we have feature a "Delete" action, the page could appear to have errors, as referenced in our support channel.

I'm proposing that we could improve the examples to show more neutral dialogs, launched by a primary button, with neutral text such as "Open Basic Modal." We could include a dedicated "Delete Item" example as well, if we wanted to show that example specifically.

category

---

## Checklist

- [ ] code has been documented
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](https://workday.github.io/canvas-kit/?path=/story/welcome-dev-docs-api-pattern-guidelines--page)

### CODE REVIEW

**Comment by emroller16** [comment]
Just created a draft PR! 

My thought here is to to change the first two examples to be more positive / neutral, while leaving the other examples as is, since "Delete" is a pretty common action that would open a modal. 

But, now on our documentation site, our examples will be a mix of neutral and destructive buttons to open dialogs, hopefully making the page look a bit more complete & reducing confusion around errors.

**Comment by mannycarrera4** [comment]
Just noticed that the full overflow example probably doesn't have enough text, can we throw in some extra text or lorem ipsum?

File: modules/react/modal/stories/examples/WithoutCloseIcon.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/WithoutCloseIcon.tsx:12]: I'm guessing `PrimaryButton` was used in this example at some point and then later removed? No need for it now since it's not currently being used.
```suggestion
import {DeleteButton} from '@workday/canvas-kit-react/button';
```

File: modules/react/modal/stories/examples/Basic.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/Basic.tsx:33]: I feel like it would be nice if the primary and secondary buttons corresponded to real-world actions rather than the generic `Primary Action` and `Secondary Action`, much like how you have `Acknowledge` and `Cancel` in the [Custom Focus](https://github.com/Workday/canvas-kit/pull/1621/files#diff-ab1bb73b5ce0a94b70381fae7f8cd55ec96167dbcf780c33a5abfd1b784755c3R35) example. Thoughts?

File: modules/react/modal/stories/examples/BodyOverflow.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/BodyOverflow.tsx:26]: Same feedback here as @mannycarrera4 had for the [Full Overflow](https://github.com/Workday/canvas-kit/pull/1621#issuecomment-1256391240) example: can we add more body text here to guarantee (more or less) the overflow is triggered for folks who view the example? Maybe we need to show a longer license 😄

File: modules/react/modal/stories/examples/BodyOverflow.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/BodyOverflow.tsx:49]: Same feedback here as I had for the Basic example: what do you think about replacing `Primary Action` and `Secondary Action` with less generic actions?

File: modules/react/modal/stories/examples/FullOverflow.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/FullOverflow.tsx:49]: Same deal re: generic actions.

**Review by jamesfan — COMMENTED** [review]
@emroller16 Thanks for taking the time to improve our examples! I do like the variety you've added -- it's nice that everything is no longer a red Delete button. I left a few comments below regarding some minor tweaks we could make to improve the examples further.

File: modules/react/modal/stories/examples/Basic.tsx
Comment (emroller16) [reply modules/react/modal/stories/examples/Basic.tsx:33]: Makes sense to me! I'll change it!

File: modules/react/modal/stories/examples/Basic.tsx
Comment (mannycarrera4) [inline modules/react/modal/stories/examples/Basic.tsx:33]: Let's change this back to live outside of `Modal.Body` and have the same padding and marginTop as before

**Review by mannycarrera4 — CHANGES_REQUESTED** [review]

File: modules/react/modal/stories/examples/FullOverflow.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/FullOverflow.tsx:71]: Same as above (`marginBottom: 0` to the last paragraph in the `Modal.Body`).
```suggestion
            <p style={{marginBottom: 0}}>
              Aliquet enim tortor at auctor urna nunc id cursus metus. Leo urna molestie at
```

**Review by jamesfan — COMMENTED** [review]
@emroller16 Changes look good! I added a few really tiny nits to clean up the spacing in the two overflow examples. This PR should be good to merge once these fixes are in + [Manny's request above](https://github.com/Workday/canvas-kit/pull/1621#discussion_r986020662) to move the `HStack` footer in the Basic story outside of the `Modal.Body`.

File: modules/react/modal/stories/examples/Basic.tsx
Comment (jamesfan) [reply modules/react/modal/stories/examples/Basic.tsx:33]: Good catch!

@emroller16 the `HStack` in the Basic story should follow the `Modal.Body` (rather than living inside it) and have `spacing="s" padding="xxs" marginTop="xxs"` (just like the `HStack` footers in the rest of the Modal examples).

Comment (emroller16) [reply modules/react/modal/stories/examples/Basic.tsx:33]: @mannycarrera4 thanks for this! I'm noticing that the basic example also uses `<Box />` inside, while some of the others don't. Should that stay?

Comment (mannycarrera4) [reply modules/react/modal/stories/examples/Basic.tsx]: Box for text is fine, the footer piece should use the HStack!

File: modules/react/modal/stories/examples/Basic.tsx
Comment (jamesfan) [inline modules/react/modal/stories/examples/Basic.tsx:32]: This should be the last change we need before this PR gets merged in. This will create the desired `24px` of vertical space between the `Modal.Body` and the `HStack` footer (see the desired spacing in #1727).
```suggestion
          <HStack spacing="s" padding="xxs" marginTop="xxs">
            <Modal.CloseButton as={PrimaryButton} onClick={handleAcknowledge}>
              Acknowledge
            </Modal.CloseButton>
```

**Review by jamesfan — CHANGES_REQUESTED** [review]
One last change (a missing margin) and then this should be good to merge!

**Review by jamesfan — APPROVED** [review]
Looks good! 👍

**Review by mannycarrera4 — APPROVED** [review]
nice work!!

**Pull request merged**
